lib/rules/amd-function-arity.js

```diff
@@ -30,10 +34,35 @@ module.exports = function (context) {
         return node.type === "FunctionExpression";
     }
 
+    function findAllowedExtraDependencies() {
+        var optionNotUndefined = context.options && context.options[0] && context.options[0].allowExtraDependencies;
+
+        if ((optionNotUndefined && context.options[0].allowExtraDependencies === true)) {
+            allowExtraDependencies = true;
+            allowedExtraDependencies = false;
+        } else if (optionNotUndefined) {
+            allowExtraDependencies = false;
+            allowedExtraDependencies = context.options[0].allowExtraDependencies;
+        } else {
+            allowExtraDependencies = false;
+            allowedExtraDependencies = false;
+        }
+    }
+
     function requireHasCallback(node) {
         return node.arguments.filter(isFunctionExpression).length;
     }
 
+    function getUnassignedPaths(dependencyNodes, callbackParams) {
+        return dependencyNodes.slice(callbackParams.length);
```

Comment:
Actually I believe the way slice works is if you pass in a positive number it grabs all the items in the array after the index you pass in. So the code above will return all the nodes that don't have a corresponding callback.

E.g. if there are 6 dependencyNodes (`[0,1,2,3,4,5]`) and 4 callbackParams doing the above would return the final two dependencyNodes. It would compute as `dependencyNodes.slice(4);` 
and return `[4,5]`.